src/main/java/com/google/sps/servlets/AnswerQuestionServlet.java

```diff
@@ -0,0 +1,156 @@
+    JsonObject jsonObj = new Gson().fromJson(request.getReader(), JsonObject.class);
+
+    String userId = getUserId(jsonObj.get("idToken").getAsString());
+
+    String gameInstanceId = jsonObj.get("gameInstanceId").getAsString();
+
+    String gameId = jsonObj.get("gameId").getAsString();
+
+    String questionId = jsonObj.get("questionId").getAsString();
+
+    String answerId = jsonObj.get("answerId").getAsString();
+
+    Firestore firestoreDb = (Firestore) this.getServletContext().getAttribute("firestoreDb");
+
+    DocumentReference gameInstanceDocRef = firestoreDb.collection("gameInstance").document(gameInstanceId);
+
+    boolean isQuestionActive = isQuestionActive(questionId, gameInstanceDocRef);
+
+    if (!isQuestionActive) {
+      return;
+    }
+
+    DocumentReference questionDocRef = firestoreDb.collection("games").document(gameId).collection("questions")
+        .document(questionId);
+
+    String correctAnswerId = getCorrectAnswer(questionId, questionDocRef);
+
+    if (answerId.equals(correctAnswerId)) {
+      System.out.println("The answer is correct");
+
+      addPoints(userId, gameInstanceDocRef, firestoreDb);
+    } else {
+      System.out.println("The answer is incorrect");
+    }
+
+    // GameInstanceDao dao = (GameInstanceDao)
+    // this.getServletContext().getAttribute("gameInstanceDao");
+
+    // dao.createNewGameInstance(jsonObj.get("idToken").getAsString(),
+    // jsonObj.get("gameId").getAsString());
+  }
+
+  private String getUserId(String idToken) {
+    FirebaseAuth firebaseAuth = (FirebaseAuth) this.getServletContext().getAttribute("firebaseAuth");
+
+    FirebaseToken decodedToken = null;
+    String userId = null;
+
+    try {
+      decodedToken = firebaseAuth.verifyIdToken(idToken);
+      userId = decodedToken.getUid();
+    } catch (FirebaseAuthException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    }
+
+    return userId;
+  }
+
+  private boolean isQuestionActive(String questionId, DocumentReference gameInstanceDocRef) {
+
+    ApiFuture<DocumentSnapshot> future = gameInstanceDocRef.get();
+
+    DocumentSnapshot document;
+    try {
+      document = future.get();
+
+      Map<String, Object> data = document.getData();
+
+      if (data.get("currentQuestion").equals(questionId)) {
+        System.out.println("Answer valid");
+        return true;
+      }
+
+    } catch (InterruptedException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    } catch (ExecutionException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
+    }
+
+    System.out.println("Answer invalid");
+
+    return false;
+  }
+
+  // Right now it only takes one single correct answer
+  private String getCorrectAnswer(String questionId, DocumentReference questionDocRef) {
+
+    String correctAnswerId = null;
+
+    ApiFuture<QuerySnapshot> future = questionDocRef.collection("answers").whereEqualTo("correct", true).get();
```

Comment:
All db querys or actions should be done from the daos, for example in here a getCorrectAnswer in game